src/lib/vocab/space/spaceHelpers.server.ts

```diff
@@ -1,28 +1,85 @@
 import {unwrap, type Result} from '@feltcoop/util';
 
-import type {AuthorizedServiceRequest} from '$lib/server/service';
-import type {CreateSpaceParams} from '$lib/app/eventTypes';
+import type {CreateSpaceParams, CreateSpaceResponse} from '$lib/app/eventTypes';
 import type {Space} from '$lib/vocab/space/space';
 import type {Entity} from '$lib/vocab/entity/entity';
 import type {DirectoryEntityData} from '$lib/vocab/entity/entityData';
-import {CreateSpaceService} from '$lib/vocab/space/spaceServices';
+import type {ApiResult} from '$lib/server/api';
+import {Logger} from '@feltcoop/util/log.js';
+import {blue, gray} from '$lib/server/colors';
+import type {Repos} from '$lib/db/Repos';
+
+const log = new Logger(gray('[') + blue('spaceHelpers.server') + gray(']'));
 
 //TODO change CreateSpace event to batched & refactor this away?
 export const createSpaces = async (
-	serviceRequest: AuthorizedServiceRequest,
 	serviceParams: CreateSpaceParams[],
+	repos: Repos,
 ): Promise<
 	Result<{value: {spaces: Space[]; directories: Array<Entity & {data: DirectoryEntityData}>}}>
 > => {
 	const spaces: Space[] = [];
 	const directories: Array<Entity & {data: DirectoryEntityData}> = [];
 	// TODO can this be safely batched? at what concurrency? or maybe make a batched repo method?
 	for (const params of serviceParams) {
+		const {community_id, name, view, path, icon} = params;
 		const {space, directory} = unwrap(
-			await CreateSpaceService.perform({...serviceRequest, params}), // eslint-disable-line no-await-in-loop
+			await createSpace(community_id, name, view, path, icon, repos), // eslint-disable-line no-await-in-loop
 		);
 		spaces.push(space);
 		directories.push(directory);
 	}
 	return {ok: true, value: {spaces, directories}};
 };
+
+export const createSpace = async (
+	community_id: number,
+	name: string,
+	view: string,
+	path: string,
+	icon: string,
```

Comment:
I think we should use the params object directly here. Pulling it apart looks more verbose and error prone, like in the future we  may change `icon` to be optional or we may add a property, these call sites know too much about its internals.

```suggestion
	params: CreateSpaceParams,
```